fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -513,18 +535,22 @@ private <M extends Message> CompletableFuture<FDBStoredRecord<M>> loadExistingRe
 
     @Nonnull
     private <M extends Message> FDBStoredRecord<M> serializeAndSaveRecord(@Nonnull RecordSerializer<M> typedSerializer, @Nonnull final FDBStoredRecordBuilder<M> recordBuilder,
-                                                                          @Nonnull final RecordMetaData metaData, @Nullable FDBStoredSizes oldSizeInfo) {
+                                                                          @Nonnull final RecordMetaData metaData, @Nullable FDBStoredSizes oldSizeInfo, final boolean isDryRun) {
```

Comment:
Why is this method called at all if `isDryRun` is true?

It seems like it would be less invasive to instead have a method to generate the new `FDBStoredRecord<M>` and then call that from `saveTypedRecord`. It also appears like the hard part here is mainly calculating the `SizeInfo`. Arguably, we could just say that dry run updates are not guaranteed to have accurate `SizeInfo` information

Reply:
this method is called when `isDryRun` = true to set `SizeInfo`.

Reply:
Yeah. I think it would be better, though, if this were only called if `isDryRun` is false, and we find a different way of calculating the `SizeInfo` information (or say that we don't care about getting the `SizeInfo` right, given that dry updates aren't really about testing if the size metrics collection logic here in the Record Layer is accurate or not). Ideally, it would seem like we want only one place in the serialize pipeline that needs to check the value of `isDryRun`. (My main concern here being that we either missed a spot, or that someone in the future modifying this code will miss a spot, and that's harder to do if all of the places that _actually_ write to the database are coalesced into a single method.)